vmm/src/lib.rs

```diff
@@ -280,9 +282,11 @@ pub fn start_vmm_thread(
         })
         .map_err(Error::VmmThreadSpawn)?;
 
+    // The VMM thread is started, we can start serving HTTP requests
     if let Some(http_path) = http_path {
-        // The VMM thread is started, we can start serving HTTP requests
-        api::start_http_thread(http_path, http_api_event, api_sender, seccomp_action)?;
+        api::start_http_path_thread(http_path, http_api_event, api_sender, seccomp_action)?;
+    } else if let Some(http_fd) = http_fd {
+        api::start_http_fd_thread(http_fd, http_api_event, api_sender, seccomp_action)?;
```

Comment:
If we always used fd could we remove some syscalls from the seccomp filter for the http thread?

Reply:
We should be able to remove socket, bind and listen I think from the filter on the http thread if we only used the fd.

Reply:
Maybe something to think about in a follow-up

Reply:
Will track with #2534